dev/requirements.txt

```diff
@@ -19,6 +19,7 @@ coverage
 
 # Linter
 mypy
+git+https://github.com/typeddjango/pytest-mypy-plugins.git@b0020061f48e85743ee3335bd62a3a608d17c6bd
```

Comment:
This was being used [in CI](https://github.com/apache/spark/blob/4bba110d9b66af8c0bafd26283ea92c733d76b53/.github/workflows/build_and_test.yml#L468) but it was missing from here, so I added it in.

In a future PR, I want to reorganize things so that CI references the requirements file. I don't think it's good that we have dependencies specified here and then specified again inside our CI script.